src/components/TrackingProvider.js

```diff
@@ -0,0 +1,15 @@
+import { Component } from 'preact';
+
+export default class TrackingProvider extends Component {
+    getChildContext() {
+        const { onTrack } = this.props;
+
+        return { track: onTrack };
+    }
+
+    render() {
```

Comment:
What happens when tracking provider have more childrens?
And why tracking provider is responsible for rendering it? Could that be avoided?

Reply:
Preact always returns `children` as an array. in opposite to React which has it's own abstraction called `React.Children` which may be a single element or an array.

From v16.0 React allows to render an array of elements, Preact hasn't implemented it (yet?) so it still requires a single element to be returned from `render`.

in case of multiple children we may just wrap in with eg.`<div>{ children }</div>` but we don't need it now.

Reply:
and no, it's not possible to skip rendering here, as providers renders an element which has the `context` passed in.